Create Ethereum guides (#8757)
ethereum/ethereum-org-website: src/content/guides/how-to-revoke-token-access/index.md

---
title: How to revoke smart contract access to your crypto funds
description: A how to guide on revoking exploitative smart contract token access
lang: en
---

> # How to: Revoke smart contract access to your crypto funds

Reviewer: Is this prefix necessary? It's not a convention we use.

```suggestion
# Revoke smart contract access to your crypto funds
```

Author: "how to" is a language used by users in survey responses and i think indicates mental model of a guide, i would recommend keeping it

> With this guide you will learn how to view a list of all smart contracts you have allowed access to your funds and also how to cancel them. 

Reviewer: ```suggestion
This guide will teach you how to view a list of all smart contracts you have allowed access to your funds and how to cancel them.
```

Reviewer: Also, 'access to your funds' or 'access to your account' 🤔 it gets muddy because you can grant access to only specific tokens I guess...

> Sometimes malicious developers build backdoors into smart contracts that will allow access to the funds of unaware users who interact with their projects. What often happens is that such platforms ask the user for permission to spend an **unlimited number of tokens**.

Reviewer: Unlimited token access isn't necessarily malicious, but it is lazy and stupid. Careful not to make the point that unlimited tokens === hacker.

Author: yeah good point maybe we should expand here why this is done (lower the number of transactions/gas on the user's side) but yeah the practice is kinda weird and i dont want to overcomplicate the article

> Sometimes malicious developers build backdoors into smart contracts that will allow access to the funds of unaware users who interact with their projects. What often happens is that such platforms ask the user for permission to spend an **unlimited number of tokens**.

Reviewer: ```suggestion
Sometimes hackers build backdoors into smart contracts to gain access to the funds of unaware users who interact with the smart contract. What often happens is that such platforms ask the user for permission to spend an **unlimited number of tokens**.
```

Reviewer: Thoughts on 'hackers' vs 'malicious developers'? It's not strictly accurate, but it's a simpler mental model that might make more sense generally IMO.

Author: not sure i agree here, on a personal level i imagine hacker to be a person exploiting some thirdparty software like hacking a bridge while malicious developer is someone who writes the software with the purpose of attracting and exploiting users.

> Once a platform has unlimited access, they can spend all your tokens even if you have withdrawn your funds from their platform into your wallet. Malicious actors can still access your funds and withdraw them into their wallet with no recovery option left for you. 

Reviewer: ```suggestion
Once a platform has unlimited access rights to a token on your wallet, they can spend all those tokens even if you have withdrawn your funds from their platform into your wallet. Malicious actors can still access your funds and withdraw them into their wallets with no recovery option left for you.
```

> The only protection is not to use untested new projects or regularly revoke access. So how to do it?

Reviewer: ```suggestion
The only protections are to refrain from using untested new projects or regularly revoke access. So, how do you do that?
```

## Step 1: Use revoke access tools

> There are several websites that let you view and revoke smart contracts connected to your address. Visit the website and connect your wallet:

Reviewer: ```suggestion
Several websites let you view and revoke smart contracts connected to your address. Visit the website and connect your wallet:
```

- [Ethallowance](https://ethallowance.com/) (Ethereum)
- [Etherscan](https://etherscan.io/tokenapprovalchecker) (Ethereum)
- [Cointool](https://cointool.app/approve/eth) (multiple networks)
- [Revoke](https://revoke.cash/) (multiple networks)
- [Unrekt](https://app.unrekt.net/) (multiple networks)
- [EverRevoke](https://everrise.com/everrevoke/) (multiple networks)

## Step 2: Connect your wallet

Once you are on the website, click on “Connect wallet”. The website should prompt you to connect your wallet.  

> Make sure you are using the right network in your wallet and on the website. You will only see smart contracts related to the network selected. For example: if you are connected to Ethereum, you will only see Ethereum contracts and not contracts from other chains like Layer 2s.

Reviewer: - active voice
- we should specify Ethereum Mainnet, as that is what most wallets with multiple network options specify 
- giving an example of another network here (Polygon) seems better to me than being generic. I don't think this will upset anyone in this instance :-)

```suggestion
Make sure you use the same network in your wallet and website. You will only see smart contracts related to the network selected. For example, if you connect to Ethereum Mainnet, you will only see Ethereum contracts, not contracts from other chains such as Polygon.
```

## Step 3: Select a smart contract you wish to revoke

> Now you should see all the contracts that are allowed access to your tokens and their potential spending limit.  Locate the one you wish to terminate.

Reviewer: ```suggestion
You should see all the contracts that are allowed access to your tokens and their spending limit. Find the one you wish to terminate.
```

> If you do not know which contract to choose, you can revoke all of them. This should not create any major issues for you. It will just mean that the next time you visit any project and wish to interact with it, you will have to grant a new set of permissions. 

Reviewer: ```suggestion
If you do not know which contract to choose, you can revoke all of them. It won't create any problems for you, but you will have to grant a new set of permissions the next time you interact with any of these contracts.
```

## Step 4: Revoke access to your funds

Once you click on revoke, you should see a new transaction suggestion in your wallet. This is to be expected. You will have to pay the fee for the cancelation to be successful. Depending on the network this can take from a minute to several to be processed.

We advise to refresh the revoking tool after few minutes and connect your wallet again to double check if revoked contract has disappeared from the list. 

> <mark> We reccomend that you never allow projects unlimited access to your tokens and that you revoke all token allowance access regularly every month. </mark>

Reviewer: Either regularly or every month, but using both together doesn't make sense.

```suggestion
<mark>We recommend you never allow projects unlimited access to your tokens and revoke all token allowance access regularly.</mark>
```

```suggestion
<mark>We recommend you never allow projects unlimited access to your tokens and revoke all token allowance access every month.</mark>
```

> ### FAQ:

Reviewer: h2

> **Does revoking token access also terminate staking, pooling, lending etc?**

Reviewer: these should be h3s IMO

> - No it will not affect any of your DEFI strategies. You will remain your positions and keep getting rewards etc.

Reviewer: These shouldn't use lists if they are single statements.

Author: @minimalsm do we have any plug and  play component for FAQs here or should this be just plain text?

Author: or @wackerow what would be your suggestion? just text without any bullet point?

> - No it will not affect any of your DEFI strategies. You will remain your positions and keep getting rewards etc.

Reviewer: ```suggestion
No, it will not affect any of your DeFi strategies. You will remain in your positions and keep getting rewards etc.
```

**Is disconnecting a wallet from a project the same as removing permission to use my funds?**

> - No, if you disconnect your wallet from the project you can still be drained of your tokens if you have granted token allowance permission. You need to revoke that access.

Reviewer: ```suggestion
- No—if you disconnect your wallet from the project, you've granted token allowance permissions, they can still use those tokens. You need to revoke that access.
```

**When will the contract permission expire?**

> - There are no expiration deadlines. As long as the permission is active it can be used, even years afterwards.

Reviewer: ```suggestion
- There are no expiration dates on contract permissions. If you grant contract permissions, they can be used, even years after they're granted.
```

---

> title: How to: Revoke smart contract access to your crypto funds

Reviewer 2: ```suggestion
title: How to revoke smart contract access to your crypto funds
```

description: A how to guide on revoking exploitative smart contract token access
lang: en
---

> # How to: Revoke smart contract access to your crypto funds

Reviewer 2: ```suggestion
# How to revoke smart contract access to your crypto funds
```
